src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -659,8 +778,7 @@ def get_observation(self, event: Event) -> pd.DataFrame:
 
         # not all simulants with a guardian are eligible to be dependents
         # need to know income of dependents in past year for this
-        dependent_w2 = df_w2[df_w2["simulant_id"].isin(df_eligible_dependents.index.unique())]
-        last_year_income = dependent_w2.groupby("simulant_id")["income"].sum()
+        last_year_income = self.w2_observer.income_last_year.groupby("simulant_id").sum()
```

Comment:
This seems like it is dependent on this observer running _after_ the W2 observer. Is that guaranteed by the order of `sub_components` above, or something else?

Reply:
> This seems like it is dependent on this observer running after the W2 observer. Is that guaranteed by the order of sub_components above, or something else?

Correct, I have added a comment about this subtle point.

Reply:
It is not guaranteed. We can guarantee it by settings a priority on the listener, but a better solution is to set last year's income on time-step cleanup so it's guaranteed to happen first (time-step cleanup happens before collect metrics).

Reply:
I will use the `time_step__cleanup` approach, it seems like the safest and clearest way.